app/src/main/java/com/fedorov/alex/whattheweatheristoday/DataCache.java

```diff
@@ -0,0 +1,53 @@
+package com.fedorov.alex.whattheweatheristoday;
+
+public class DataCache {
+    private static int imageRes = -1;
+    private static String city, temp;
```

Comment:
Это же итак синглтон, то есть все его объекты в единичном экземпляре.
Точнее так: этот объект всегда один, и все его поля тоже в одном экземпляре.
Зачем тут статик?

```suggestion
    private String city;
    private String temp;
```